provider/reprovider.go

```diff
@@ -424,7 +410,7 @@ func (s *reprovider) run() {
 				return
 			}
 
-			err := s.reprovide(s.ctx, false)
+			err := s.Reprovide(s.ctx)
```

Comment:
`s.Reprovide` is already selecting over `s.ctx` so there is no additional context to watch when calling `s.Reprovide`. So pass in background context instead.
```suggestion
			err := s.Reprovide(context.Background())
```